### PR TITLE
FEAT: CAP-36 - Fully implemented PICK TEAM (team-selection) page

### PR DESCRIPTION
**Pull request by abhijeetpraveen**
## Description

**JIRA Ticket:** [CAP-36](https://mfslcapstone.atlassian.net/browse/CAP-36)

**Summary:**

This PR is for the **Team Selection** page. 

- On this page, users will set their fantasy team lineup for the upcoming fantasy week (to be done in the backend still).
- Players can be substituted from the bench (with error checking), the bench position can also be changed (for automatic substitutions).
- When a player is clicked a modal is open (allowing users to make substitutions)
- A player's info can also be viewed using the `View Information` button. A modal is opened with their past results (with stats) and upcoming games. Their total points for the season as well as team, position and name info is also visible.
- Also (Sorry 😢 should be another ticket and PR) but added code in this ticket for navigating to a 404 error page not found when the user tries to write a url that doesn't exist.

**Screenshots/Examples:**
**The page:**

League

**Clicking on a starting XI player:**

Screenshot 2024-03-25 at 9 16 25 PM

**Clicking on a substitute:**
Screenshot 2024-03-25 at 9 16 59 PM

**Viewing Player Information:**

- **Results:**
Screenshot 2024-03-25 at 9 20 56 PM

- **Fixtures:**
Screenshot 2024-03-25 at 9 21 10 PM

**Page Not Found:**
Screenshot 2024-03-25 at 9 32 43 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-functional change which does not add or remove functionality)

## Checklist

- [x] My branch and PR name follow the [naming conventions](https://mfslcapstone.atlassian.net/wiki/spaces/SD/pages/4358145/Development+Naming+Conventions) the team has established.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules. (If not, this should be a draft PR)
- [x] I have [rebased](https://mfslcapstone.atlassian.net/wiki/spaces/SD/pages/4456479/How+To%3A+Rebase+Branch+Onto+Main) my branch onto main before opening this PR, and resolved any merge conflicts.
- [x] I have ensured that new and existing integrations work with my changes.

**Dependencies (if applicable):**

### In package.json

1. `classnames`: Only used once but works properly
2. `react-color`: Not sure why this is here, can't find where I used it or if at all. Keeping it here coz we might still want to use it sometime later in the future.
3. `react-toastify`: For error and success notifications
4. `sass`: For scss files (only used once but works properly)

---

[CAP-36]: https://mfslcapstone.atlassian.net/browse/CAP-36?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by seb8stien1** [comment]
Not to be that guy, but youre supposed to create PRs before they reach over 1000 additions

File: src/Pages/TeamSelectionPage/TeamSelectionPage.tsx
Comment (seb8stien1) [inline src/Pages/TeamSelectionPage/TeamSelectionPage.tsx:1]: It doesnt even open the diffs by default cuz this is so big lmao

File: src/api/api.ts [hunk 1]
```diff
@@ -2,14 +2,13 @@ import axios, { Method } from 'axios';
 
 // Base Axios instance
 const api = axios.create({
-    baseURL: 'https://mfsl.loca.lt',
+    baseURL: 'http://localhost:8080',
```
Comment (seb8stien1) [inline src/api/api.ts:5]: Should we make this an env file so it can be overriden on the deployed version

**Review by seb8stien1 — APPROVED** [review]

File: src/api/api.ts (hunk 1 above)
Comment (abhijeetpraveen) [reply src/api/api.ts:5]: Will address this at the end of the project

**Comment by abhijeetpraveen** [comment]
Hey @rooshnie23, I know @seb8stien1  has approved it but can you also take a look at the code please. Just glance over it coz I know its huge. 🙏🏾

**Review by rooshnie23 — APPROVED** [review]
Skimmed over this massive PR, I think it should be good. Hopefully, this code doesn't break anything.

**Comment by abhijeetpraveen** [comment]
> Skimmed over this massive PR, I think it should be good. Hopefully, this code doesn't break anything.

It shouldn't, I rebased it already

**Pull request merged**